### PR TITLE
Numpy 2.0 support

### PR DESCRIPTION
**Pull request by aganders3**
NumPy 2.0 is set to be released on June 16. I've been adding support in a couple other projects so figured I'd try to help out here if you're interested.

The process here was mainly to run the [ruff plugin](https://numpy.org/devdocs/numpy_2_0_migration_guide.html#ruff-plugin) and fix what it caught. I used `np.float64` where I needed to pick a float precision, because that seemed more common in your code.

I also added the optional dependencies I found when running tests as extras, so you can install them with `pip install phidl[all]` (or `[test]`). Then I updated a few places that were causing `DeprecationWarning`s.

### CODE REVIEW

File: tests/test_geometry.py [hunk 1]
```diff
@@ -229,8 +230,9 @@ def test_truetype():
 
 @pytest.mark.skipif(sys.version_info < (3, 0), reason="unicode test requires python3")
 def test_unicode():
-    _ = pytest.importorskip(
-        "phidl.font", reason="Testing of ttf/otf fonts requires the freetype package."
+    pytest.importorskip(
+        "phidl.font", reason="Testing of ttf/otf fonts requires the freetype package.",
+        exc_type=ImportError,
```
Comment (aganders3) [inline tests/test_geometry.py:235]: This could go in a separate PR, and this might not be the correct solution.
https://docs.pytest.org/en/8.2.x/deprecations.html#import-or-skip-import-error

File: phidl/device_layout.py [hunk 2]
```diff
@@ -354,7 +354,8 @@ def _line_distances(points, start, end):
         return np.linalg.norm(points - start, axis=1)
 
     vec = end - start
-    cross = np.cross(vec, start - points)
+    vec2 = start - points
```
Comment (aganders3) [inline phidl/device_layout.py:357]: This is not a good name :/

**Comment by amccaugh** [comment]
Thank you for the PR!  I've made a few modifications to satisfy the tests but otherwise looks great!

**Comment by aganders3** [comment]
Thanks! I'm not sure why pre-commit was not catching these issues locally for me but oh well.

Pour one out for Python 3.7 support.

**Comment by amccaugh** [comment]
No idea what was going wrong on the 3.7 but it was EOL anyway!
